core/systems/crypto/package_verifier.gd

```diff
@@ -48,3 +48,18 @@ func get_hash(data: PackedByteArray, type: HashingContext.HashType = HashingCont
 ## Get the hash of the given data as a hex encoded string
 func get_hash_string(data: PackedByteArray, type: HashingContext.HashType = HashingContext.HASH_SHA256) -> String:
 	return get_hash(data, type).hex_encode()
+
+
+## Returns the hash of the file at the given path
+func get_file_hash(path: String, type: HashingContext.HashType = HashingContext.HASH_SHA256) -> PackedByteArray:
+	if not FileAccess.file_exists(path):
+		logger.warn("File does not exist at path: " + path)
+		return PackedByteArray()
+	var data := FileAccess.get_file_as_bytes(path)
+	
+	return get_hash(data, type)
+
+
+## Get the hash of the file at the given path as a hex encoded string
+func get_file_hash_string(path: String, type: HashingContext.HashType = HashingContext.HASH_SHA256) -> String:
```

Comment:
This provides very little security, so I'm not sure what the point of this is? Corruption validation? A bad actor editing the binary could easily edit the text file. Should we host the per-version hash in an online database similar to how crankshaft did it?

Reply:
Right, all this part does is check to see if the files are corrupted and check to see if the file in the update archive needs to be extracted. The signature part is what provides the security.